storage_backend_sftp/components/sftp_adapter.py

```diff
@@ -35,7 +36,12 @@ def sftp(backend):
     account = backend._get_keychain_account()
     password = account._get_password()
     transport = paramiko.Transport((backend.sftp_server, backend.sftp_port))
-    transport.connect(username=backend.sftp_login, password=password)
+    if backend.sftp_auth_method == 'pwd':
+        transport.connect(username=backend.sftp_login, password=password)
+    elif backend.sftp_auth_method == 'ssh_key':
+        ssh_key_buffer = StringIO(password)
+        private_key = paramiko.RSAKey.from_private_key(ssh_key_buffer)
```

Comment:
While RSA is the default, both ssh and paramiko support other formats, such as ECDSA. In the paramiko client class, they just try each in sucession until one works, we could do the same:

```python
def load_ssh_key(ssh_key_buffer):
    for pkey_class in (RSAKey, DSSKey, ECDSAKey, Ed25519Key):
        try:
            return pkey_class.from_private_key(ssh_key_buffer)
        except SSHException:
            ssh_key_buffer.seek(0) # reset the buffer "file"
    raise Exception("Invalid ssh private key")